socket.go

```diff
@@ -305,7 +305,8 @@ func (st SocketType) String() string {
 type SocketOptionLevel int32
 
 const (
-	SocketLevel SocketOptionLevel = iota
+	SocketLevel SocketOptionLevel = 0
+	TcpLevel    SocketOptionLevel = 6
```

Comment:
I know that 6 was chosen because it's the value of `IPPROTO_TCP`, but again there's a chance that WasmEdge [adds levels](https://github.com/WasmEdge/WasmEdge/blob/d649efad4a8d33e5302fdc24c6e8e823b08e9369/thirdparty/wasi/api.hpp#L1716) that clashes with this. When we receive a value of 6, we won't know whether it's a WasmEdge module or whether it's a module that uses `IPPROTO_TCP`.